### PR TITLE
# Blog app - creating a data model 

### PR DESCRIPTION
**Pull request by JEFFKAY16**
## Blog app - creating a data model 

In this update I did the following:
- I reference the [ERD diagram here](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png)
- Built a project schema with the following: 
    - Created and ran the necessary migration files.
    - Table and column names match the ERD diagram.
    - The photo for the users table is a link to an image
    - Foreign keys are included.
    - All columns that are foreign keys have a corresponding index.

### CODE REVIEW

File: db/migrate/20220429201735_create_users.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+class CreateUsers < ActiveRecord::Migration[7.0]
+
+  def change
+
+    create_table :users do |t|
+      t.string :Name
+      t.string :Photo
+      t.text :Bio
+      t.integer :PostsCounter
```
Comment (bimbolabuari) [inline db/migrate/20220429201735_create_users.rb:9]: As part of Ruby's best practice, I suggest you use `snake_case` to name your variables.
image

File: db/migrate/20220429202100_create_posts.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+      t.integer :AuthorId
+      t.string :Title
+      t.text :Text
+      t.integer :CommentsCounter
+      t.integer :LikesCounter
+
+      t.timestamps
+    end
+
+    add_foreign_key :posts, :users, column: :AuthorId
```
Comment (bimbolabuari) [inline db/migrate/20220429202100_create_posts.rb:15]: As part of Ruby's best practice, I suggest you use `snake_case` to name your variables.
image

Kindly check for other occurrences 👍

**Review by bimbolabuari — CHANGES_REQUESTED** [review]
Hi @JEFFKAY16, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: db/migrate/20220429201735_create_users.rb (hunk 1 above)
Comment (JEFFKAY16) [reply db/migrate/20220429201735_create_users.rb:9]: @bimbolabuari Hi, 
Thanks for your review. Please note that the requirements says, they must be created `exactly` as shown in the ERD diagram.

File: db/migrate/20220429202100_create_posts.rb (hunk 2 above)
Comment (JEFFKAY16) [reply db/migrate/20220429202100_create_posts.rb:15]: @bimbolabuari Hi, 
Thanks for your review. Please note that the requirements says, they must be created `exactly` as shown in the ERD diagram.

File: db/migrate/20220429201735_create_users.rb (hunk 1 above)
Comment (mmsesay) [inline db/migrate/20220429201735_create_users.rb:9]: - [ ] I will suggest you implement the changes required by the previous reviewer. I understand the point you are making about the requirement. However, the requirement is trying to point out that, if a column is called `Name` you should not define it as `username` or something else. It doesn't mean you should title the columns with capitalizations.

As the previous reviewer mentioned, snake_case is for methods and variables. While CamelCase is for classes and modules.

Please ensure to update all migrations where you have capitalize column names.

**Review by mmsesay — CHANGES_REQUESTED** [review]
Hi @JEFFKAY16, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: db/migrate/20220429201735_create_users.rb (hunk 1 above)
Comment (JEFFKAY16) [reply db/migrate/20220429201735_create_users.rb:9]: @mmsesay 
So why wasn't that done in the ERD diagram? I will not make this change, if anything this should have been optional

Comment (JEFFKAY16) [reply db/migrate/20220429201735_create_users.rb:9]: @mmsesay Done

**Review by Cathella — APPROVED** [review]
## 🔥 Status: Approved 🟢

### Hi @JEFFKAY16 👋🏻, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


Cheers and Happy coding!👏👏👏

rosycheeks-mochi-peach


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**